tests/FsToolkit.ErrorHandling.IcedTasks.Tests/CancellableTaskValidationCE.fs

```diff
@@ -0,0 +1,525 @@
+module CancellableTaskValidationCETests
+
+
+#if FABLE_COMPILER
+open Fable.Mocha
+#else
+open Expecto
+#endif
+
+open TestHelpers
+open FsToolkit.ErrorHandling
+open FsToolkit.ErrorHandling.Operator.Result
+
+let ``CancellableTaskValidationCE return Tests`` =
+    testList "CancellableTaskValidationCE return Tests" [
+        testCaseAsync "Return string"
+        <| async {
+            let data = "Foo"
+            let! actual = cancellableTaskValidation { return data }
+            Expect.equal actual (Result.Ok data) "Should be ok"
+        }
+    ]
+
+let ``CancellableTaskValidationCE return! Tests`` =
+    testList "CancellableTaskValidationCE return! Tests" [
+        testCaseAsync "Return Ok result"
+        <| async {
+            let data = Result.Ok "Foo"
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual (data) "Should be ok"
+        }
+        testCaseAsync "Return Error result"
+        <| async {
+            let innerData = "Foo"
+            let! expected = CancellableTaskValidation.error innerData
+            let data = Result.Error innerData
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual expected "Should be error"
+        }
+        testCaseAsync "Return Ok Choice"
+        <| async {
+            let innerData = "Foo"
+            let data = Choice1Of2 innerData
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual (Result.Ok innerData) "Should be ok"
+        }
+        testCaseAsync "Return Error Choice"
+        <| async {
+            let innerData = "Foo"
+            let! expected = CancellableTaskValidation.error innerData
+            let data = Choice2Of2 innerData
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual expected "Should be error"
+        }
+        testCaseAsync "Return Ok Validation"
+        <| async {
+            let innerData = "Foo"
+            let! data = CancellableTaskValidation.ok innerData
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual (Result.Ok innerData) "Should be ok"
+        }
+        testCaseAsync "Return Error Validation"
+        <| async {
+            let innerData = "Foo"
+            let! expected = CancellableTaskValidation.error innerData
+            let data = CancellableTaskValidation.error innerData
+            let! actual = cancellableTaskValidation { return! data }
+            Expect.equal actual expected "Should be ok"
+        }
+    ]
+
+
+let ``CancellableTaskValidationCE bind Tests`` =
+    testList "CancellableTaskValidationCE bind Tests" [
```

Comment:
This is a good start but I think we're also going to need tests for:

- Task
- Task<'T>
- Any awaitable (like Task.Yield()) (this should cover ValueTask/ValueTask<'T> as well)
- Async
- All these variants with Result/Validation as well.  


Yeah this is gonna hurt.